apis/identity/v1alpha1/iamuserpolicyattachment_types.go

```diff
@@ -28,11 +28,19 @@ type IAMUserPolicyAttachmentParameters struct {
 	// PolicyARN is the Amazon Resource Name (ARN) of the IAM policy you want to
 	// attach.
 	// +immutable
-	PolicyARN string `json:"policyArn"`
+	PolicyARN string `json:"policyArn,omitempty"`
 
-	// UserName presents the name of the IAMUser.
+	// PolicyARNRef references an IAMPolicy to retrieve its Policy ARN.
+	// +optional
+	PolicyARNRef *runtimev1alpha1.Reference `json:"policyArnRef,omitempty"`
+
+	// PolicyARNSelector selects a reference to an IAMPolicy to retrieve its
+	// Policy ARN
 	// +optional
-	UserName *string `json:"userName,omitempty"`
+	PolicyARNSelector *runtimev1alpha1.Selector `json:"policyArnSelector,omitempty"`
+
+	// UserName presents the name of the IAMUser.
+	UserName string `json:"userName,omitempty"`
```

Comment:
Wondering if maybe we should mark this as `// +immutable` since we don't actually update if it does get changed. What do you think?